### ISSUE TITLE
Configuration bootstrapping strategies

### ISSUE THREAD

**Issue by ffbit** [ffbit/solr-maven-plugin#23]
Implement Apache Solr configuration bootstrapping strategies to:
- provide default working demo configuration stuff, as Apache Solr deliveries do, for further tuning
- support an existent/legacy Apache Solr configuration environment

**Comment by ffbit** [comment]
For first approximation it would be:
- bootstrap once
- bootstrap everytime
- never bootstrap
- bootstrap if (a conditional option)

**Comment by ffbit** [comment]
Implemented in the #25 pull request for:
- bootstrap once as **ONCE**
- bootstrap every time as **EVERY_TIME** (default)
- never bootstrap as **NEVER**

Usage examples:

**ONCE**

```
<configuration>
    <bootstrappingStrategy>ONCE</bootstrappingStrategy>
</configuration>
```

**EVERY_TIME**

```
<configuration>
    <bootstrappingStrategy>EVERY_TIME</bootstrappingStrategy>
</configuration>
```

**NEVER**

```
<configuration>
    <bootstrappingStrategy>NEVER</bootstrappingStrategy>
</configuration>
```
